pkg/instrumentation/sdk.go

```diff
@@ -472,7 +472,11 @@ func (i *sdkInjector) createResourceMap(ctx context.Context, otelinst v1alpha1.I
 	k8sResources[semconv.K8SPodNameKey] = pod.Name
 	k8sResources[semconv.K8SPodUIDKey] = string(pod.UID)
 	k8sResources[semconv.K8SNodeNameKey] = pod.Spec.NodeName
-	k8sResources[semconv.ServiceInstanceIDKey] = createServiceInstanceId(ns.Name, fmt.Sprintf("$(%s)", constants.EnvPodName), pod.Spec.Containers[index].Name)
+	podName := pod.Name
+	if podName == "" {
+		podName = fmt.Sprintf("$(%s)", constants.EnvPodName)
+	}
+	k8sResources[semconv.ServiceInstanceIDKey] = createServiceInstanceId(ns.Name, podName, pod.Spec.Containers[index].Name)
```

Comment:
Instead of adding a special case here to account for a different special case later added in `injectCommonSDKConfig`, could we not always set `OTEL_RESOURCE_ATTRIBUTES_POD_NAME` from the downward API, and then always use that value in the resource map? That should be simpler and more consistent.

Reply:
Yes, it is better.

Reply:
Fixed @swiatekm